### ISSUE TITLE
dotnet40 installation on clean 64bit prefix hangs

### ISSUE THREAD

**Issue by Zemogiter** [Winetricks/winetricks#1088]
After the installation window for .net framework verifies the setup file it hangs at "Installing .NET Framework 4 Client Profile". Using the latest version of winetricks and wine 3.14 on Ubuntu 18.04 x64
Here is the log (some text like "This packet (packetname) may not work in x64 prefix" was translated from Polish):
```
------------------------------------------------------
You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20180815-next - sha256sum: 1664876f765d9187d28d5fe745588d814f4c673bb8d31c6f326aee34dc145dc9 with wine-3.14 and WINEARCH=win64
Executing w_do_call dotnet461
Executing load_dotnet461 
------------------------------------------------------
This packet (dotnet461) may not work correctly with 64-bit installation. 32-bit prefix may work better.
------------------------------------------------------
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Bypassing error in wine 34803 
------------------------------------------------------
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/system32/mscoree.dll
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/syswow64/mscoree.dll
Executing w_do_call dotnet46
Executing load_dotnet46 
------------------------------------------------------
This packet (dotnet46) may not work correctly with 64-bit installation. 32-bit prefix may work better.
------------------------------------------------------
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Mono does not appear to be installed.
------------------------------------------------------
------------------------------------------------------
Bypassing error in wine 34803 
------------------------------------------------------
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/system32/mscoree.dll
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/syswow64/mscoree.dll
Executing w_do_call dotnet45
Executing load_dotnet45 
------------------------------------------------------
This packet (dotnet45) may not work correctly with 64-bit installation. 32-bit prefix may work better.
------------------------------------------------------
Executing cabextract -q -q -v
cabextract version 1.6
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Mono does not appear to be installed.
------------------------------------------------------
------------------------------------------------------
Bypassing error in  wine 34803 
------------------------------------------------------
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/system32/mscoree.dll
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/syswow64/mscoree.dll
Executing w_do_call dotnet40
Executing load_dotnet40 
------------------------------------------------------
This packet (dotnet40) may not work correctly with 64-bit installation. 32-bit prefix may work better.
------------------------------------------------------
------------------------------------------------------
dotnet40 does not yet fully work or install on wine.  Caveat emptor.
------------------------------------------------------
------------------------------------------------------
Bypassing error in wine 42701 
------------------------------------------------------
------------------------------------------------------
On 64-bit, you'll run into https://bugs.winehq.org/show_bug.cgi?id=42701 (missing api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll.RoGetParameterizedTypeInstanceIID
------------------------------------------------------
Executing w_do_call remove_mono
Executing load_remove_mono 
------------------------------------------------------
Mono does not appear to be installed.
------------------------------------------------------
------------------------------------------------------
Bypassing error in wine 34803 
------------------------------------------------------
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
reg: System was unable to find the value or registry key
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/system32/mscoree.dll
Executing rm -f /home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers/dosdevices/c:/windows/syswow64/mscoree.dll
Executing w_do_call winxp
Executing load_winxp 
Operation ended successfuly
Setting Windows version to winxp
Executing wine regedit C:\windows\Temp\_winxp\set-winver.reg
Executing wine64 regedit C:\windows\Temp\_winxp\set-winver.reg
------------------------------------------------------
Running /home/jonasz/.PlayOnLinux//wine/linux-amd64/3.14/bin//wineserver -w. This will hang until all wine processes in prefix=/home/jonasz/.PlayOnLinux//wineprefix/SpaceEngineers terminate
------------------------------------------------------
Executing cd /home/jonasz/.cache/winetricks/dotnet40
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0
000d:err:menubuilder:convert_to_native_icon error 0x80004005 creating bitmap encoder
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
0034:err:mscoree:LoadLibraryShim error reading registry key for installroot
003e:err:msiexec:custom_action_server Failed to create custom action server pipe: 231
```

**Comment by dmitry64** [comment]
I had this issue with wine 3.14 on ubuntu 18.04. I've managed to install dotnet40 with latest wine built from source. Perhaps they fixed something right after 3.14 release.

**Comment by gverm** [comment]
This was [Bug 45663](https://bugs.winehq.org/show_bug.cgi?id=45663) and it was fixed already. Fix will be included in wine-3.15.

**Comment by qwertychouskie** [comment]
@Zemogiter It seems Wine is looking for some libraries not available on your system.

    000d:err:wincodecs:PngEncoder_CreateInstance Failed writing PNG because unable to find libpng12.so.0

  Are you using the packages from https://wiki.winehq.org/Ubuntu?

**Comment by PietJankbal** [comment]
I think this ticket can be closed, in wine-3.17 this works again. (the regressions in msi have been fixed) Could you try?
